File: froom-backend/src/main/java/com/froom/backend/user/controller/UserController.java
package com.froom.backend.user.controller;

import com.froom.backend.authorization.model.command.RegisterUserCommand;
import com.froom.backend.user.model.domain.User;
import com.froom.backend.user.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController("user")
@RequiredArgsConstructor
public class UserController {

    private final UserRepository userRepository;

//    private final UserRepository userRepository;
//    private final PasswordEncoder passwordEncoder;

//    @PostMapping("/register")
//    public void registerUser(@RequestBody @Validated RegisterUserCommand command) {
//        if (userRepository.findByEmail(command.getEmail()).isPresent()) {
//            throw new IllegalArgumentException("Email is already registered");
//        }
//
//        // Create a new User entity and set its properties
//        User user = new User();
//        user.setFirstName(command.getFirstName());
//        user.setLastName(command.getLastName());
//        user.setEmail(command.getEmail());
//        user.setPassword(passwordEncoder.encode(command.getPassword()));
//
//        // Save the user to the database
//        userRepository.save(user);
//    }

    @GetMapping("/me")
    public void getMyself() {

    }

    @GetMapping("/users")
    public Iterable<User> getAllUsers() {
//        return userRepository.findAll();
        List<User> users = userRepository.findAll();
        System.out.println("Users: " + users); // Log the users
        return users;
    }

    @PutMapping
    public void updateUser() {

    }

    @DeleteMapping
    public void deleteUser() {

    }
}

File: froom-backend/src/main/kotlin/com/froom/froombackend/user/controller/UserController.kt
package com.froom.froombackend.user.controller

import com.froom.froombackend.authorization.util.toUser
import com.froom.froombackend.user.model.command.RegisterUserCommand
import com.froom.froombackend.user.model.command.UpdatePasswordCommand
import com.froom.froombackend.user.model.command.UpdateUserCommand
import com.froom.froombackend.user.model.dto.UserDto
import com.froom.froombackend.user.service.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/user")
class UserController (
    private val userService: UserService
)  {

    @GetMapping()
    fun getUser(authentication: Authentication): ResponseEntity<UserDto> {
        return ResponseEntity.ok(userService.getUser(authentication.toUser()))
    }

    @PostMapping("/register")
    fun registerUser(@RequestBody @Valid command: RegisterUserCommand): ResponseEntity<UserDto> {
        return ResponseEntity(userService.registerUser(command), HttpStatus.CREATED)
    }

    @PutMapping()
    fun updateUser(@RequestBody @Valid command: UpdateUserCommand, authentication: Authentication): ResponseEntity<UserDto> {
        return ResponseEntity(userService.updateUser(command, authentication.toUser()), HttpStatus.ACCEPTED)
    }

    @PutMapping("/password")
    fun updatePassword(@RequestBody @Valid command: UpdatePasswordCommand, authentication: Authentication): ResponseEntity<UserDto> {
        return ResponseEntity(userService.updatePassword(command, authentication.toUser()), HttpStatus.ACCEPTED)
    }

    @DeleteMapping()
    fun deleteUser(authentication: Authentication): ResponseEntity<UserDto> {
        return ResponseEntity(userService.removeUser(authentication.toUser()), HttpStatus.OK)
    }

}